psydac/fem/lst_preconditioner.py

```diff
@@ -0,0 +1,293 @@
+#---------------------------------------------------------------------------#
+# This file is part of PSYDAC which is released under MIT License. See the  #
+# LICENSE file or go to https://github.com/pyccel/psydac/blob/devel/LICENSE #
+# for full license details.                                                 #
+#---------------------------------------------------------------------------#
+import  numpy as np
+
+from    functools                       import lru_cache
+
+from    scipy.sparse                    import dia_matrix
+
+from    sympde.topology                 import elements_of, Line, ScalarFunctionSpace
+from    sympde.topology.datatype        import SpaceType
+from    sympde.expr                     import integral, BilinearForm
+
+from    psydac.fem.projectors           import DirichletProjector
+from    psydac.linalg.basic             import IdentityOperator, LinearOperator
+from    psydac.linalg.block             import BlockVectorSpace, BlockLinearOperator
+from    psydac.linalg.direct_solvers    import BandedSolver
+from    psydac.linalg.kron              import KroneckerLinearSolver, KroneckerStencilMatrix
+from    psydac.linalg.stencil           import StencilVectorSpace
+
+@lru_cache
```

Comment:
The module `functools` is part of the Python standard library, so it should be imported before the third-party packages. Further, we can remove some useless whitespace here

```suggestion
from   functools                    import lru_cache

import numpy as np
from   scipy.sparse                 import dia_matrix

from   sympde.topology              import elements_of, Line, ScalarFunctionSpace
from   sympde.topology.datatype     import SpaceType
from   sympde.expr                  import integral, BilinearForm

from   psydac.linalg.basic          import IdentityOperator, LinearOperator
from   psydac.linalg.block          import BlockVectorSpace, BlockLinearOperator
from   psydac.linalg.direct_solvers import BandedSolver
from   psydac.linalg.kron           import KroneckerLinearSolver, KroneckerStencilMatrix
from   psydac.linalg.stencil        import StencilVectorSpace
from   psydac.fem.projectors        import DirichletProjector

@lru_cache
```